packages/bff/src/services/toolsService.ts

```diff
@@ -0,0 +1,393 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import {
+  ApiKey,
+  ClientAssertion,
+  ConsumerKey,
+  FailedValidation,
+  SuccessfulValidation,
+  validateClientKindAndPlatformState,
+  validateRequestParameters,
+  verifyClientAssertion,
+  verifyClientAssertionSignature,
+} from "pagopa-interop-client-assertion-validation";
+import {
+  AgreementId,
+  ApiError,
+  ClientId,
+  EServiceId,
+  ItemState,
+  PurposeId,
+  TenantId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { getAllFromPaginated, WithLogger } from "pagopa-interop-commons";
+import {
+  agreementApi,
+  authorizationApi,
+  bffApi,
+  catalogApi,
+  purposeApi,
+} from "pagopa-interop-api-clients";
+import {
+  AgreementProcessClient,
+  CatalogProcessClient,
+  PagoPAInteropBeClients,
+} from "../providers/clientProvider.js";
+import { BffAppContext } from "../utilities/context.js";
+import {
+  activeAgreementByEserviceAndConsumerNotFound,
+  clientAssertionPublicKeyNotFound,
+  ErrorCodes,
+  eserviceDescriptorNotFound,
+  missingActivePurposeVersion,
+  multipleAgreementForEserviceAndConsumer,
+  purposeIdNotFoundInClientAssertion,
+} from "../model/domain/errors.js";
+import { TokenGenerationValidationStepFailure } from "../../../api-clients/dist/bffApi.js";
+
+export function toolsServiceBuilder(clients: PagoPAInteropBeClients) {
+  return {
+    async validateTokenGeneration(
+      clientId: string | undefined,
+      clientAssertion: string,
+      _clientAssertionType: string,
+      _grantType: string,
+      ctx: WithLogger<BffAppContext>
+    ): Promise<bffApi.TokenGenerationValidationResult> {
+      const { errors: parametersErrors } = validateRequestParameters({
+        client_assertion: clientAssertion,
+        client_assertion_type:
+          "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
+        grant_type: "client_credentials",
```

Comment:
Leftover from testing? Here you should probably pass the two unused function params

Reply:
Yes, in any case that is the only accepted string there. Should we check the user input if it matches that string and if not throw an error?

Reply:
In Scala if I'm not wrong the two params are passed to the client assertion - I don't know if the check should be implemented or not, let's see what other reviewers say, but it's weird that we have two unused params coming from the front-end. If that's actually the case, let's add a comment to mention why they are unused and to maybe remove them from the API spec in the future